src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1063,6 +1274,57 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
             return node;
         }
 
+        public override void VisitPattern(BoundPattern pattern)
+        {
+            Visit(pattern);
+            Split();
+        }
+
+        public override BoundNode VisitConstantPattern(BoundConstantPattern node)
+        {
+            Visit(node.Value);
+            return null;
+        }
+
+        public override BoundNode VisitDeclarationPattern(BoundDeclarationPattern node)
+        {
+            Visit(node.VariableAccess);
+            Visit(node.DeclaredType);
+            return null;
+        }
+
+        public override BoundNode VisitDiscardPattern(BoundDiscardPattern node)
+        {
+            return null;
+        }
+
+        public override BoundNode VisitITuplePattern(BoundITuplePattern node)
+        {
+            VisitAll(node.Subpatterns);
+            return null;
+        }
+
+        public override BoundNode VisitRecursivePattern(BoundRecursivePattern node)
+        {
+            Visit(node.DeclaredType);
+            if (!node.Deconstruction.IsDefault)
```

Comment:
>if (!node.Deconstruction.IsDefault) [](start = 12, length = 35)

Consider moving this check into `VisitAll`, since the same check is used in several instances.